src/codeu/chat/client/commandline/Chat.java

```diff
@@ -14,310 +14,384 @@
-
-      clientContext.conversation.showAllConversations();
-
-    } else if (token.equals("c-select")) {
-
-      selectConversation(lineScanner);
+      // Find the first user with the given name and return a user context
+      // for that user. If no user is found, the function will return null.
+      private UserContext findUser(String name) {
+        for (final UserContext user : context.allUsers()) {
+          if (user.user.name.equals(name)) {
+            return user;
+          }
+        }
+        return null;
+      }
+    });
 
-    } else if (token.equals("m-add")) {
+    // Now that the panel has all its commands registered, return the panel
+    // so that it can be used.
+    return panel;
+  }
 
-      if (!clientContext.user.hasCurrent()) {
-        System.out.println("ERROR: Not signed in.");
-      } else if (!clientContext.conversation.hasCurrent()) {
-        System.out.println("ERROR: No conversation selected.");
-      } else {
-        if (!tokenScanner.hasNext()) {
-          System.out.println("ERROR: Message body not supplied.");
+  private Panel createUserPanel(final UserContext user) {
+
+    final Panel panel = new Panel();
+
+    // HELP
+    //
+    // Add a command that will print a list of all commands and their
+    // descriptions when the user enters "help" while on the user panel.
+    //
+    panel.register("help", new Panel.Command() {
+      @Override
+      public void invoke(Scanner args) {
+        System.out.println("USER MODE");
+        System.out.println("  c-list");
+        System.out.println("    List all conversation that the current user can interact with.");
+        System.out.println("  c-add <title>");
+        System.out.println("    Add a new conversation with the given title and join it as the current user.");
+        System.out.println("  c-join <title>");
+        System.out.println("    Join the conversation as the current user.");
+        System.out.println("  info");
+        System.out.println("    Display all info for the current user");
+        System.out.println("  back");
+        System.out.println("    Go back to ROOT MODE.");
+        System.out.println("  exit");
+        System.out.println("    Exit the program.");
+      }
+    });
+
+    // C-LIST (list conversations)
+    //
+    // Add a command that will print all conversations when the user enters
+    // "c-list" while on the user panel.
+    //
+    panel.register("c-list", new Panel.Command() {
+      @Override
+      public void invoke(Scanner args) {
+        for (final ConversationContext conversation : user.conversations()) {
+          System.out.format(
+              "CONVERSATION %s (%s)\n",
+              conversation.conversation.title,
+              conversation.conversation.id);
+        }
+      }
+    });
+
+    // C-ADD (add conversation)
+    //
+    // Add a command that will create and join a new conversation when the user
+    // enters "c-add" while on the user panel.
```

Comment:
"when the user enters "c-add" while on the user panel" is not really needed.  The command name is the first thing you see on the next source line.